WordPress/src/main/java/org/wordpress/android/ui/publicize/PublicizeErrorFragment.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.ui.publicize
+
+import android.app.AlertDialog
+import android.app.Dialog
+import android.os.Bundle
+import android.view.ContextThemeWrapper
+import androidx.annotation.StringRes
+import androidx.fragment.app.DialogFragment
+import org.wordpress.android.R
+import org.wordpress.android.R.style
+import org.wordpress.android.ui.WPWebViewActivity
+import org.wordpress.android.ui.publicize.PublicizeConstants.PUBLICIZE_FACEBOOK_SHARING_SUPPORT_LINK
+import java.lang.NullPointerException
+
+class PublicizeErrorFragment : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        arguments?.let {
+            it.getInt(REASON_RES_ID).let { reasonResId ->
+                // the custom dialog behavior here is specific to a user who is trying to share to Facebook
+                // but does not have any available Facebook Pages.
+                if (reasonResId == R.string.sharing_facebook_account_must_have_pages) {
+                    return AlertDialog.Builder(
+                            ContextThemeWrapper(
+                                    activity,
+                                    style.Calypso_Dialog
+                            )
+                    ).apply {
+                        setTitle(R.string.dialog_title_sharing_facebook_account_must_have_pages)
+                        setMessage(R.string.sharing_facebook_account_must_have_pages)
+                        setPositiveButton(R.string.ok) { dialog, _ -> dialog.dismiss() }
+                        setNegativeButton(R.string.learn_more) { _, _ ->
+                            WPWebViewActivity.openURL(
+                                    activity,
+                                    PUBLICIZE_FACEBOOK_SHARING_SUPPORT_LINK
+                            )
+                        }
+                    }.create()
+                } else {
```

Comment:
I put both dialogs here since they are both displaying an error/failure. Is this approach fine? Double-checking to ensure I am not somehow putting too much responsibility in the class. I just thought it wouldn't make sense to have two of these since the main difference is just the contents of the dialog being displayed.

Reply:
Yeah, that's fine in my opinion :)